assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_student_view_submission.vm

```diff
@@ -44,6 +44,45 @@ function submitform(id)
 	}
 }
 	
+/*For the cancel button - if the user made progress, we need them to confirm that they want to discard their progress*/
+function confirmDiscardOrSubmit()
+{
+	var inlineProgress = false;
+	var ckEditor = CKEDITOR.instances["$name_submission_text"];
+	if (ckEditor)
+	{
+		inlineProgress = CKEDITOR.instances["$name_submission_text"].checkDirty();
+	}
+	var attachmentsModified = $!new_attachments;
+	var showDiscardDialog = inlineProgress || attachmentsModified;
+	var submitPanel = document.getElementById("submitPanel");
+	var confirmationDialogue = document.getElementById("confirmationDialogue");
+	if (showDiscardDialog)
+	{
+		submitPanel.setAttribute('style', 'display:none;');
+		confirmationDialogue.removeAttribute('style');
+	}
+	else
+	{
+		cancelProceed();
+	}
+}
+
+function cancelProceed()
+{
+	document.addSubmissionForm.onsubmit();
+	document.addSubmissionForm.option.value='cancel'; 
+	document.addSubmissionForm.submit(); 
+}
+
+function undoCancel()
+{
+	var submitPanel = document.getElementById("submitPanel");
+	var confirmationDialogue = document.getElementById("confirmationDialogue");
+	submitPanel.removeAttribute('style');
+	confirmationDialogue.setAttribute('style', 'display:none;');
+}
+
```

Comment:
I don't think we should continue to bloat these .vm files with JavaScript. These should be put into the existing assignments.js file, or if they are specific to this one page, maybe introduce a new JavaScript file.

Also, it is best practise to namespace JavaScript functions to avoid collisions with 3rd party libraries. You can see an example of this in the user tool, or like I've done in https://github.com/sakaiproject/sakai/pull/468